net/src/client/http/mod.rs

```diff
@@ -0,0 +1,62 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use futures03::future::BoxFuture;
+use surf::http::{self, Response, Uri};
+use surf::middleware::{Body, HttpClient as SurfHttpClient};
+use surf::Client;
+
+type Request = http::Request<Body>;
+
+/// A surf-alike HTTP client that additionally supports proxies (HTTP(S), SOCKS4 and SOCKS5)
+#[derive(Clone, Debug)]
+pub struct WitnetHttpClient {
+    client: Arc<isahc::HttpClient>,
+}
+
+impl WitnetHttpClient {
+    /// Create a new `WitnetHttpClient`
+    pub fn new(proxy: &Option<String>) -> Self {
+        let proxy_address = proxy.as_ref().map(|x| Uri::from_str(x.as_str()).unwrap());
+        let client = Arc::new(
+            if let Some(proxy_address) = proxy_address {
+                isahc::HttpClient::builder().proxy(proxy_address)
+            } else {
+                isahc::HttpClient::builder()
+            }
+            .build()
+            .unwrap(),
+        );
+
+        Self { client }
+    }
+
+    /// Turn this `WitnetHttpClient` into a `surf`-compatible client
+    pub fn as_surf_client(&self) -> Client<Self> {
+        Client::with_client(self.clone())
```

Comment:
Move the clone to the caller, so this function can also be used without cloning.

```suggestion
    pub fn as_surf_client(self) -> Client<Self> {
        Client::with_client(self)
```

Reply:
Ah nevermind, it's just an Arc clone so it doesn't matter